modules/core/src/main/java/org/apache/ignite/IgniteCluster.java

```diff
@@ -442,17 +443,36 @@
      * Checks Ignite grid is active or not active.
      *
      * @return {@code True} if grid is active. {@code False} If grid is not active.
+     * @deprecated Use {@link #state()} instead.
      */
+    @Deprecated
     public boolean active();
 
     /**
      * Changes Ignite grid state to active or inactive.
      *
      * @param active If {@code True} start activation process. If {@code False} start deactivation process.
      * @throws IgniteException If there is an already started transaction or lock in the same thread.
+     * @deprecated Use {@link #state(ClusterState)} instead.
      */
+    @Deprecated
     public void active(boolean active);
 
+    /**
+     * Gets current cluster state.
+     *
+     * @return Current cluster state.
+     */
+    public ClusterState state();
+
+    /**
+     * Changes current cluster state to given {@code newState} cluster state.
+     *
+     * @param newState New cluster state.
+     * @throws IgniteException If there is an already started transaction or lock in the same thread.
```

Comment:
what does **lock in the same thread** mean ?

Reply:
I've reused our exchange mechanism and extend the cluster activation process. This Javadoc was copied from `IgniteCluster#active(boolean)` method because this is the same logic.